onnx/onnx.in.proto

```diff
@@ -135,6 +202,20 @@ message AttributeProto {
   repeated bytes strings = 9;         // list of UTF-8 strings
   repeated TensorProto tensors = 10;  // list of tensors
   repeated GraphProto graphs = 11;    // list of graph
+
+// #if ONNX-ML
+  optional TupleAttribute        tuple    = 100; // A single tuple value
+  optional RecordAttribute       record   = 101; // A single record value
+  optional NullableAttribute     nullable = 102; // A single nullable value
+  optional SparseTensorAttribute sparse   = 103; // A single sparse tensor
+  optional TypeProto             typea    = 104; // A single type
+
+  repeated TupleAttribute        tuples    = 105; // List of tuple values
+  repeated RecordAttribute       records   = 106; // List of record values
+  repeated NullableAttribute     nullables = 107; // List of nullable values
+  repeated SparseTensorAttribute sparses   = 108; // List of sparse tensors
+  repeated TypeProto             types     = 109; // List of types
+// #endif
```

Comment:
What about Maps? I guess Sequences are omitted since AttributeProto provides a built-in support via the "repeated" fields. Maps could be represented as Sequences of Tuples, but I wonder it it helps to be explicit. On a related note, I guess nothing here imposes the requirement of "homogeneity" … e.g., that when we have "repeated TupleAttribute", all the tuples must have the same type. I guess we imply this, if we want to be able to type these attributes.